Sources/MongoSwift/BSON/Document+Sequence.swift

```diff
@@ -93,8 +236,34 @@ public class DocumentIterator: IteratorProtocol {
         return values
     }
 
+    // uses an iterator to copy (key, value) pairs of the provided document from range [startIndex, endIndex) into a new
+    // document. starts at the startIndex-th pair and ends at the end of the document or the (endIndex-1)th index, 
+    // whichever comes first. 
+    internal static func subsequence(of doc: Document, startIndex: Int = 0, endIndex: Int = Int.max) -> Document {
```

Comment:
shouldn't we be using [bson_copy_to_excluding_noinit](http://mongoc.org/libbson/current/bson_copy_to_excluding_noinit.html) here instead of doing this all in Swift?

Reply:
That's a bummer, maybe we should open a ticket in the C driver for this. The way we are doing this is pretty expensive

Reply:
I've opened [CDRIVER-2829](https://jira.mongodb.org/browse/CDRIVER-2829) for discussion.